azkaban-web-server/src/main/java/azkaban/flowtrigger/TriggerInstanceProcessor.java

```diff
@@ -88,22 +91,56 @@ private void executeFlowAndUpdateExecID(final TriggerInstance triggerInst) {
-    return String.format(FAILURE_EMAIL_SUBJECT, flowFullName, this.emailer.getAzkabanName());
+    return String.format(FAILURE_EMAIL_SUBJECT, triggerInstance.getFlowId(), triggerInstance
+        .getProjectName(), this.emailer.getAzkabanName());
   }
 
-  private String generateFailureEmailBody(final TriggerInstance triggerInstance) {
-    final String triggerInstFullName =
-        triggerInstance.getProjectName() + "." + triggerInstance.getFlowId();
-    return String.format(FAILURE_EMAIL_BODY, triggerInstFullName);
+  private EmailMessage createFlowTriggerEmailMessage(final TriggerInstance triggerInst) {
+    final EmailMessage message = this.emailer.createEmailMessage(generateFailureEmailSubject
+        (triggerInst), "text/html", triggerInst.getFailureEmails());
+    final SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+
+    message.addAllToAddress(triggerInst.getFailureEmails());
+    message.setMimeType("text/html");
+    message.println("<table>");
+    message.println("<tr><td>Start Time</td><td>");
+    message.println("<tr><td>" + sdf.format(new Date(triggerInst.getStartTime())) + "</td><td>");
+
+    message.println("<tr><td>End Time</td><td>");
+    message.println("<tr><td>" + sdf.format(new Date(triggerInst.getEndTime())) + "</td><td>");
+    message.println("<tr><td>Duration</td><td>"
+        + Utils.formatDuration(triggerInst.getStartTime(), triggerInst.getEndTime())
+        + "</td></tr>");
+    message.println("<tr><td>Status</td><td>" + triggerInst.getStatus() + "</td></tr>");
+    message.println("</table>");
+    message.println("");
+    final String executionUrl = this.emailer.getAzkabanURL()+ "/executor?triggerinstanceid="
+            + triggerInst.getId();
+
+    message.println("<a href=\"" + executionUrl + "\">" + triggerInst.getFlowId()
+        + " Flow Trigger Instance Link</a>");
+
+    message.println("");
+    message.println("<h3>Cancelled Dependencies</h3>");
+
+    for (final DependencyInstance depInst : triggerInst.getDepInstances()) {
+      if (depInst.getStatus() == Status.CANCELLED) {
+        message.println("<table>");
+        message.println("<tr><td>Dependency Name: " + depInst.getDepName() + "</td><td>");
+        message
+            .println("<tr><td>Cancellation Cause: " + depInst.getCancellationCause() + "</td><td>");
+        message.println("</table>");
+      }
+    }
+
+    return message;
   }
 
   private void sendFailureEmailIfConfigured(final TriggerInstance triggerInstance) {
     final List<String> failureEmails = triggerInstance.getFailureEmails();
     if (!failureEmails.isEmpty()) {
-      this.emailer.sendEmail(failureEmails, generateFailureEmailSubject(triggerInstance),
-          generateFailureEmailBody(triggerInstance));
+      final EmailMessage message = this.createFlowTriggerEmailMessage(triggerInstance);
+      this.emailer.sendEmail(message, true, "email message " + message.getBody());
```

Comment:
Isn't `operation = "email message " + message.getBody()` a bit too much to be included in info-level logging (see Emailer.java)? I'd suggest for example `"flow trigger failure email"` instead (result => `logger.info("Sent " + "flow trigger failure email")`). Any way if you're including the entire email body content on purpose, fine by me..